lib/ssh_data/public_key/skecdsa.rb

```diff
@@ -34,8 +35,26 @@ def rfc4253
         )
       end
 
-      def verify(signed_data, signature)
-        raise UnsupportedError, "SK-ECDSA verification is not supported."
+      def verify(signed_data, signature, **opts)
+        opts = DEFAULT_SK_VERIFY_OPTS.merge(opts)
+        unknown_opts = opts.keys - DEFAULT_SK_VERIFY_OPTS.keys
+        raise UnsupportedError, "Verification options #{unknown_opts.inspect} are not supported." unless unknown_opts.empty?
+
+        sig_algo, raw_sig, sk_flags, blob = build_signing_blob(application, signed_data, signature)
+        self.class.check_algorithm!(sig_algo, curve)
+
+        openssl_sig = self.class.openssl_signature(raw_sig)
+        digest = DIGEST_FOR_CURVE[curve]
+
+        result = openssl.verify(digest.new, openssl_sig, blob)
+
+        if opts[:user_presence_required] && (sk_flags & SK_FLAG_USER_PRESENCE != SK_FLAG_USER_PRESENCE)
```

Comment:
Maybe contrived..but is it possible both flags are passed? If possible, the code as written would hit/check the presence condition and the `elsif` would cause us to never check the verification bit. If anything, it feels like like user verification should take precedence. Or..maybe we just check them all with any `elsif`(ing)?

Reply:
> but is it possible both flags are passed? 

Looking at What OpenSSH does, it checks the flags independently of one another https://github.com/openssh/openssh-portable/blob/7a7c69d8b4022b1e5c0afb169c416af8ce70f3e8/sshconnect2.c#L1266-L1273

One does not imply the other. Though in practice I was unable to generate a key that had UV but not UP without resorting to manually assembling bits.

I'll update the code to make that more clear.